Scripts/Runtime/Job/Util/JobHandleExtension.cs

```diff
@@ -31,5 +37,17 @@ public static bool IsDependencyOf(this JobHandle job, JobHandle candidateJob)
         {
             return JobHandle.CheckFenceIsDependencyOrDidSyncFence(job, candidateJob);
         }
+
+        /// <summary>
+        /// Determines whether two <see cref="JobHandle"/> instances are equal without boxing.
+        /// </summary>
+        /// <param name="job1">The first <see cref="JobHandle"/> compare.</param>
+        /// <param name="job2">The second <see cref="JobHandle"/> compare.</param>
+        /// <returns>True if the <see cref="JobHandle"/>s are the same.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe bool Equals_NoBox(this in JobHandle job1, in JobHandle job2)
```

Comment:
Should use `ref` because `JobHandle` is not a `readonly` struct and so an defensive copy will be made.

```suggestion
        public static unsafe bool Equals_NoBox(this ref JobHandle job1, ref JobHandle job2)
```

Reply:
Defensive copy is only made if you interact with a method (including getters unless marked `readonly`)

Using `in` allows us to use inline returned values when calling vs `ref` that would require us to assign to a temp variable.

Reply:
Interesting.... I was wrong here, my bad.